framework/DBDD_optimized.sage

```diff
@@ -0,0 +1,210 @@
+from fpylll import *
+from fpylll.algorithms.bkz2 import BKZReduction
+
+load("../framework/load_strategies.sage")
+load("../framework/DBDD.sage")
+load("../framework/proba_utils.sage")
+
+def need_lattices_bases(fn):
+    """
```

Comment:
why plurals in the function name ?

Reply:
It is because it updated primal and dual bases, but you are right. In the context, because hint integration only requires one basis, it must be in the singular. I will rename this decorator.